### PR TITLE
Fixes catbans and cleaned up some other minor code [READY]

### LINKED ISSUE #10730: Catbanned players spawn on the START SCREEN
I dunno if it's just when I am on, but anyone that is catbanned and tries to join the game spawns on the title screen. They aren't even a catbeast and can pretty much do nothing, all naked and yet still set to their character preferences.

### PR DESCRIPTION
**Pull request by yoyobatty**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl: YoYoBatty
fix: Catbeasts no longer get randomly sent to the start area without a job!
fix: Catbans sometimes didn't always change your species, this has now been fixed.
fix: Fixed circuit logs not appearing
/:cl:

## About The Pull Request
This fixes catbans. Closes: https://github.com/HippieStation/HippieStation/issues/10730 https://github.com/HippieStation/HippieStation/issues/10892

## Why It's Good For The Game
Because it's been bothering everyone.

### CODE REVIEW

**Comment by ghost** [comment]
#### Test Merge Deployed

##### Server Instance
HippieStation

##### Revision
Origin: 0fdcc2edeeef2c89d10b61a5b2350968f8d19ea7
Pull Request: 2f61b29906c6abb36288fd9bc44e9e1204941aea
Server: fcc78812b4e28de2b0861c29bc9e1d24c820bc94

##### Comment
Many more tests to come

**Comment by Jujumatic** [comment]
now to do something horrible so we can get catbanned and test this out 👌

**Comment by ghost** [comment]
#### Test Merge Updated

##### Server Instance
HippieStation

##### Revision
Origin: c1dfd48aeab1b1e747225b65ffc39f204c1f2cfc
Pull Request: 676c411ac71029b6ed9e5e0bda53da37d96ab3e6
Server: fa8e4c477b56d2f6c16c95152a4b67fafbed3dbf

##### Comment
Many more tests to come

**Comment by ghost** [comment]
#### Test Merge Removed

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -81,8 +81,6 @@ SUBSYSTEM_DEF(job)
 			return FALSE
 		if(is_banned_from(player.ckey, rank) || QDELETED(player))
 			return FALSE
-		if((is_banned_from(player.ckey, CLUWNEBAN) || is_banned_from(player.ckey, CATBAN)) && !is_banned_from(player.ckey, SSjob.overflow_role)) // hippie start -- fixes catbans
```
Comment (Carbonhell) [inline code/controllers/subsystem/job.dm:84]: Why did you remove this?

Comment (yoyobatty) [reply code/controllers/subsystem/job.dm:84]: Causes it to break, already discussed in discord.
